avalanche/distributed/distributed_tensor.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import torch
+from torch import Tensor
+
+from avalanche.distributed import DistributedHelper
+from avalanche.distributed.distributed_value import SwitchableDistributedValue
+
+
+class DistributedTensor(SwitchableDistributedValue[Tensor, Tensor], ABC):
+    """
+    A distributed Tensor wrapper.
+
+    This abstract class is in charge of synchronizing Tensors across processes.
+
+    Child classes must override `_merge_tensors` to define how those tensors
+    should be merged.
+    """
+    def _synchronize_distributed_value(self) -> Tensor:
+        return self._merge_tensors(
+            DistributedHelper.gather_all(self.local_value))
+
+    @abstractmethod
+    def _merge_tensors(self, tensors: List[Tensor]) -> Tensor:
```

Comment:
maybe just `_merge`? I would avoid different names for each type since it's always the same operation (just different implementations).